mrmustard/math/math_interface.py

```diff
@@ -933,13 +872,20 @@ def random_symplectic(self, num_modes: int = 1, max_r: float = 1.0) -> Tensor:
         dd = self.diag(self.concat([self.exp(-r), np.exp(r)], axis=0), k=0)
         return OW @ dd @ OV
 
-    def random_orthogonal(self, num_modes: int = 1) -> Tensor:
-        """A random orthogonal matrix in :math:`O(2*num_modes)`."""
-        if num_modes == 1:
+    @staticmethod
+    def random_orthogonal(N: int) -> Tensor:
+        """A random orthogonal matrix in :math:`O(N)`."""
+        if N == 1:
+            return np.array([[1.0]])
+        return ortho_group.rvs(dim=N)
+
+    def random_unitary(self, N: int) -> Tensor:
```

Comment:
I would suggest a (very minor) refactoring here just for the sake of readability:
```suggestion
    def random_unitary(self, N: int) -> Tensor:
        """a random unitary matrix in :math:`U(N)`"""
        if N == 1:
            return self.exp(1j * np.random.uniform(size=(1, 1)))
        return unitary_group.rvs(dim=N)
```